xcvm/lib/core/src/network.rs

```diff
@@ -0,0 +1,96 @@
+use crate::abstraction::IndexOf;
+use alloc::vec::Vec;
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+
+/// Newtype for XCVM networks ID. Must be unique for each network and must never change.
+/// This ID is an opaque, arbitrary type from the XCVM protocol and no assumption must be made on
+/// how it is computed.
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+#[derive(
+	Copy,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	Debug,
+	Encode,
+	Decode,
+	TypeInfo,
+	Serialize,
+	Deserialize,
+)]
+#[repr(transparent)]
+pub struct NetworkId(pub u8);
+
+impl From<u8> for NetworkId {
+	fn from(x: u8) -> Self {
+		NetworkId(x)
+	}
+}
+
+impl From<Picasso> for NetworkId {
+	fn from(_: Picasso) -> Self {
+		Picasso::ID
+	}
+}
+
+impl From<Ethereum> for NetworkId {
+	fn from(_: Ethereum) -> Self {
+		Ethereum::ID
+	}
+}
+
+impl From<Juno> for NetworkId {
+	fn from(_: Juno) -> Self {
+		Juno::ID
+	}
+}
+
+/// The index 0 must not be used for safety purpose, we hence introduce an invalid network at this
+/// index.
+pub struct InvalidNetwork;
+/// Composable Picasso (Kusama parachain)
+pub struct Picasso;
+/// Ethereum mainnet
+pub struct Ethereum;
+/// Juno (Cosmos) mainnet
+pub struct Juno;
+
+/// List of networks supported by XCVM.
+/// /!\ The order matters and must not be changed, adding a network on the right is safe.
```

Comment:
```suggestion
// /!\ The order matters and must not be changed, adding a network on the right is safe.
```

Comments for maintainers should not be doc comments btw